### PR TITLE
tests: increased test coverage, minor bug fixes

### PR DESCRIPTION
**Pull request by ArtemShatokhin**
> [!NOTE]
> Adds broad unit/integration test coverage (FastAPI logging, MCP server, ToolFactory, BaseTool, citations, persistence), hardens tool module importing, tweaks default tool description, and updates dev deps/Makefile sync.
> 
> - **Tests**:
>   - Add comprehensive unit/integration suites:
>     - FastAPI logging middleware (request tracking, file logs, retrieval endpoint).
>     - MCP server (HTTP/stdio), auth middleware, error cases, BaseTool conversion.
>     - ToolFactory (core import/adaptation, OpenAPI, LangChain integration) and real tool execution.
>     - BaseTool behavior (schema, context, config), citation extractor utilities, UI persistence edge cases.
>   - Update test data: `tests/data/tools/sample_tool.py` to import `function_tool` from `agency_swarm`.
> - **Tools**:
>   - `ToolFactory.from_file`: guard module existence before accessing members to avoid errors on failed imports.
>   - `BaseTool.openai_schema`: simplify default description to "`<ClassName>` tool".
> - **Build/Dev**:
>   - `pyproject.toml` `tool.uv.dev-dependencies`: add FastAPI, HTTP, LangChain, and related packages for tests.
>   - `Makefile` `sync`: include `--dev` to install dev dependencies.

### CODE REVIEW

File: tests/test_agent_modules/test_tool_factory_openapi.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+                    "required": ["required_field"],
+                }
+            }
+        }
+
+        callback = ToolFactory._create_invoke_for_path("/test", "POST", {}, tool_schema, "test_func")
+
+        # Test real validation error with missing required field
+        with pytest.raises((ValidationError, Exception)):  # Should raise validation error
+            await callback(None, '{"parameters": {}}')  # Missing required_field
```
Comment (nicko-ai) [inline tests/test_agent_modules/test_tool_factory_openapi.py]: Update the validation error handling test to supply OpenAPI server metadata and expect the explicit ModelBehaviorError:

```suggestion
        # Provide minimal OpenAPI metadata so the invoke path builds a proper URL
        openapi = {"servers": [{"url": "https://api.example.com"}]}
        callback = ToolFactory._create_invoke_for_path("/test", "POST", openapi, tool_schema, "test_func")

        # Test real validation error with missing required field
        with pytest.raises(ModelBehaviorError, match="Invalid JSON input in parameters"):
            await callback(None, '{"parameters": {}}')  # Missing required_field
```

File: tests/test_agent_modules/test_tool_factory_openapi.py [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+Unit tests for ToolFactory OpenAPI and schema functionality.
+
+Tests the OpenAPI schema conversion and HTTP callback functionality with real data.
+"""
+
+import json
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from agents import FunctionTool
```
Comment (nicko-ai) [inline tests/test_agent_modules/test_tool_factory_openapi.py:11]: Import ModelBehaviorError for the OpenAPI tool factory tests to assert the precise failure type:

```suggestion
from agents import FunctionTool
from agents.exceptions import ModelBehaviorError
```

File: tests/test_agent_modules/test_tool_factory_langchain.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
+"""
+Unit tests for ToolFactory LangChain integration.
+
+Tests the LangChain tool conversion functionality with REAL LangChain tools.
+"""
+
+from unittest.mock import patch
```
Comment (nicko-ai) [inline tests/test_agent_modules/test_tool_factory_langchain.py:8]:
```suggestion
import builtins
from unittest.mock import patch
```

File: tests/test_agent_modules/test_tool_factory_langchain.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        """Test error when langchain_community is not available."""
+
+        # Create a simple dummy tool class
+        class DummyTool:
+            def run(self):
+                return "dummy"
+
+        dummy_tool = DummyTool()
+
+        # Mock import to simulate ImportError
```
Comment (nicko-ai) [inline tests/test_agent_modules/test_tool_factory_langchain.py:29]: Capture the original __import__ before patching in the LangChain tool factory test to avoid recursive mock calls when simulating missing dependencies

```suggestion
        # Mock import to simulate ImportError
        original_import = builtins.__import__
```

File: tests/test_agent_modules/test_tool_factory_langchain.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        with patch("builtins.__import__") as mock_import:
+
+            def import_side_effect(name, *args, **kwargs):
+                if name == "langchain_community.tools":
+                    raise ImportError("No module named 'langchain_community'")
+                return __import__(name, *args, **kwargs)
```
Comment (nicko-ai) [inline tests/test_agent_modules/test_tool_factory_langchain.py]:
```suggestion
                return original_import(name, *args, **kwargs)
```

**Review by bonk1t — CHANGES_REQUESTED** [review]

File: pyproject.toml [hunk 6]
```diff
@@ -66,6 +66,9 @@ dev-dependencies = [
     "pytest-asyncio>=1.0.0",
     "pytest-mock>=3.14.1",
     "ruff>=0.11.12",
+    "langchain-core>=0.1.0",
+    "langchain-experimental>=0.0.50",
+    "langchain-community>=0.0.20",
 ]
```
Comment (nicko-ai) [inline pyproject.toml:78]: Update the sync target so make ci installs all extras together with the dev dependency set, ensuring CI environments pull the FastAPI stack:

```suggestion
    "fastapi>=0.115.0",
    "uvicorn[standard]>=0.34.0",
    "ag-ui-protocol>=0.1.4",
    "aiofiles>=23.2.1",
    "filetype>=1.2.0",
    "httpx>=0.28.0",
]
```

And in the Makefile, please add this:

```
@@ -1,28 +1,28 @@
 .PHONY: sync
 sync:
-	uv sync --all-extras
+	uv sync --all-extras --dev
```

Then run `make sync` locally to regenerate uv.lock.

File: tests/integration/fastapi/test_logging_middleware.py
Comment (nicko-ai) [inline tests/integration/fastapi/test_logging_middleware.py:1]: An improvement: 
https://github.com/VRSEN/agency-swarm/pull/379/files

File: src/agency_swarm/tools/tool_factory.py [hunk 7]
```diff
@@ -354,17 +355,18 @@ def from_file(file_path: str | Path) -> list[type[BaseTool] | FunctionTool]:
...
+            if inspect.isclass(base_tool) and issubclass(base_tool, BaseTool) and base_tool is not BaseTool:
+                try:
+                    tools.append(base_tool)
+                except Exception as e:
+                    logger.error("Error adapting tool %s: %s", module_name, e)
+
+            # FunctionTool instances defined in the module
+            for obj in module.__dict__.values():
+                if isinstance(obj, FunctionTool):
+                    tools.append(obj)
```
[the remark this answers is not in the document]
Comment (ArtemShatokhin) [reply src/agency_swarm/tools/tool_factory.py:369]: Ignoring is not silent, an error will be displayed by logger. Errors are not raised intentionally.

**Review by bonk1t — APPROVED** [review]
LGTM
